cloudfoundry/provider/datasource_service_credential_binding.go

```diff
@@ -32,7 +32,10 @@ func (d *ServiceCredentialBindingDataSource) Metadata(ctx context.Context, req d
 
 func (d *ServiceCredentialBindingDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
 	resp.Schema = schema.Schema{
-		MarkdownDescription: "Gets information on Service Credential Bindings for a given service instance.",
+		MarkdownDescription: "Gets information on Service Credential Bindings for a given service instance.\n\n" +
+			"This datasource will be deprecated in a future release.\n\n" +
```

Comment:
I would propose to mark the text as follows:

Use the datasource `cloudfoundry_service_credential_binding_details` instead when you want to fetch details of a specific service credential binding of type `app` or `key`.\n\n" +
			"Use Use the datasource `cloudfoundry_service_credential_bindings` if you want to fetch all credential bindings (both `app` and `key` types) for a given service instance.",